---
title: "C6103 | Microsoft Docs"
ms.date: 11/15/2016
ms.prod: "visual-studio-dev14"
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords: 
  - "C6103"
ms.assetid: 22d1ab35-31a3-4ba9-8ef4-7a64bce66621
caps.latest.revision: 6
author: corob-msft
ms.author: corob
manager: jillfra
---
# C6103
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

warning C6103: Returning \<variable> from failed function call at line \<location>.  
  
 A successful path through the function is returning a variable that was used as an `_Out_` parameter to an internal function call that failed.  
  
 The problem might be that the called function and the calling function are not completely annotated. The called function may require `_Always_`, `_Outptr_result_nullonfailure_` (`_COM_Outptr_` for COM code), or a related annotation, and the calling function may require a `_Success_` annotation. Another possible cause for this warning is that the `_Success_` annotation on the called function is incorrect.  
  
## See Also  
 [Using SAL Annotations to Reduce C/C++ Code Defects](../code-quality/using-sal-annotations-to-reduce-c-cpp-code-defects.md)
